### PR TITLE
feat: add `math/base/special/rad2degf`

### LINKED ISSUE #2124: [RFC]: add `math/base/special/rad2degf`
### Description

This RFC proposes adding `math/base/special/rad2degf`, which would be the the single-precision equivalent for [`math/base/special/rad2deg`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/rad2deg).

```c
float stdlib_base_rad2degf( const float x )
```

### Related Issues

None.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by gunjjoshi**
Resolves #2124.

## Description

> What is the purpose of this pull request?

This pull request:

-   adds `math/base/special/rad2degf`, which would be the the single-precision equivalent for [`math/base/special/rad2deg`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/rad2deg).

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   fixes #2124

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

- Code Coverage Report:

Screenshot 2024-04-06 at 02 40 04

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/math/base/special/rad2degf/test/test.js [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+	var expected;
+	var delta;
+	var tol;
+	var er;
+	var x;
+	var r;
+	var i;
+
+	x = [
+		PI / 6.0,
```
Comment (gunjjoshi) [inline lib/node_modules/@stdlib/math/base/special/rad2degf/test/test.js:101]: One thing that I wanted to ask here is that, should we use the `float64ToFloat32` wrapper for these values, as they involve arithmetic operations ?

File: lib/node_modules/@stdlib/math/base/special/rad2degf/lib/main.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+// MODULES //
+
+var float64ToFloat32 = require( '@stdlib/number/float64/base/to-float32' );
+
+
+// VARIABLES //
+
+// 180.0 / π
+var CONST_180_DIV_PI = 57.29577951308232;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/rad2degf/lib/main.js]:
```suggestion
var CONST_180_DIV_PI = float64ToFloat32( 57.29577951308232 );
```

File: lib/node_modules/@stdlib/math/base/special/rad2degf/lib/main.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+* @example
+* var d = rad2degf( -3.141592653589793 / 4.0 );
+* // returns -45.0
+*
+* @example
+* var d = rad2degf( NaN );
+* // returns NaN
+*/
+function rad2degf( x ) {
+	return float64ToFloat32( x * float64ToFloat32( CONST_180_DIV_PI ) );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/rad2degf/lib/main.js]:
```suggestion
	return float64ToFloat32( x * CONST_180_DIV_PI );
```

File: lib/node_modules/@stdlib/math/base/special/rad2degf/lib/main.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+* @example
+* var d = rad2degf( -3.141592653589793 / 4.0 );
+* // returns -45.0
+*
+* @example
+* var d = rad2degf( NaN );
+* // returns NaN
+*/
+function rad2degf( x ) {
+	return float64ToFloat32( x * CONST_180_DIV_PI );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/rad2degf/lib/main.js]:
```suggestion
	return float64ToFloat32( float64ToFloat32( x ) * CONST_180_DIV_PI );
```

You need to mimic the input value casting of C when providing a double to a single-precision interface. Meaning, in JS, every `x` needs to be explicitly "converted" to float32.

File: lib/node_modules/@stdlib/math/base/special/rad2degf/src/main.c [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+static const float CONST_180_DIV_PI = 57.29577951308232f;
+
+/**
+* Converts an angle from radians to degrees (single-precision).
+*
+* @param x    input value (in radians)
+* @return     output value (in degrees)
+*
+* @example
+* float x = 3.141592653589793f / 2.0f
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/rad2degf/src/main.c]:
```suggestion
* float x = 3.141592653589793f / 2.0f;
```

File: lib/node_modules/@stdlib/math/base/special/rad2degf/test/test.js (hunk 1 above)
Comment (kgryte) [reply lib/node_modules/@stdlib/math/base/special/rad2degf/test/test.js:101]: Should be fine to leave as is.

File: lib/node_modules/@stdlib/math/base/special/rad2degf/test/test.js [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+		360.0
+	];
+	for ( i = 0; i < x.length; i++ ) {
+		r = rad2degf( x[ i ] );
+		er = float64ToFloat32( expected[ i ] );
+		if ( r === er ) {
+			t.equal( r, er, 'x: '+x[ i ]+'. r: '+r+'. expected: '+er+'.' );
+		} else {
+			delta = abs( r - er );
+			tol = abs( EPS * er );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/rad2degf/test/test.js]:
```suggestion
			tol = EPS * abs( er );
```

Comment (kgryte) [reply lib/node_modules/@stdlib/math/base/special/rad2degf/test/test.js]: No reason to change things up here. Should be consistent with the above.

Comment (kgryte) [reply lib/node_modules/@stdlib/math/base/special/rad2degf/test/test.js]: I don't know why this was done this way in `rad2deg`.

Comment (kgryte) [reply lib/node_modules/@stdlib/math/base/special/rad2degf/test/test.js]: I went ahead and changed upstream.

File: lib/node_modules/@stdlib/math/base/special/rad2degf/test/test.js [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+			t.equal( delta <= tol, true, 'x: '+x[ i ]+'. r: '+r+'. expected: '+er+'. delta: '+delta+'. tol: '+ tol+'.' );
+		}
+		// Negative `x`:
+		r = rad2degf( -x[ i ] );
+		er = float64ToFloat32( -expected[ i ] );
+		if ( r === er ) {
+			t.equal( r, er, 'x: '+x[ i ]+'. r: '+r+'. expected: '+er+'.' );
+		} else {
+			delta = abs( r - er );
+			tol = abs( EPS * er );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/rad2degf/test/test.js]:
```suggestion
			tol = EPS * abs( er );
```

File: lib/node_modules/@stdlib/math/base/special/rad2degf/test/test.native.js [hunk 8]
```diff
@@ -0,0 +1,174 @@
...
+			t.equal( delta <= tol, true, 'x: '+x[ i ]+'. r: '+r+'. expected: '+er+'. delta: '+delta+'. tol: '+ tol+'.' );
+		}
+		// Negative `x`:
+		r = rad2degf( -x[ i ] );
+		er = float64ToFloat32( -expected[ i ] );
+		if ( r === er ) {
+			t.equal( r, er, 'x: '+x[ i ]+'. r: '+r+'. expected: '+er+'.' );
+		} else {
+			delta = abs( r - er );
+			tol = abs( EPS * er );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/rad2degf/test/test.native.js]:
```suggestion
			tol = EPS * abs( er );
			t.equal( delta <= tol, true, 'x: '+x[ i ]+'. r: '+r+'. expected: '+er+'. delta: '+delta+'. tol: '+ tol+'.' );
		}
		// Negative `x`:
		r = rad2degf( -x[ i ] );
		er = float64ToFloat32( -expected[ i ] );
		if ( r === er ) {
			t.equal( r, er, 'x: '+x[ i ]+'. r: '+r+'. expected: '+er+'.' );
		} else {
			delta = abs( r - er );
			tol = EPS * abs( er );
```

File: lib/node_modules/@stdlib/math/base/special/rad2degf/test/test.js [hunk 9]
```diff
@@ -0,0 +1,165 @@
...
+			delta = abs( r - er );
+			tol = EPS * abs( er );
+			t.equal( delta <= tol, true, 'x: '+x[ i ]+'. r: '+r+'. expected: '+er+'. delta: '+delta+'. tol: '+ tol+'.' );
+		}
+	}
+	t.end();
+});
+
+tape( 'if provided a value greater than `~3.14e+38`, the function will underflow', function test( t ) {
+	var r = rad2degf( 3.14e+38 );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/rad2degf/test/test.js]:
```suggestion
tape( 'if provided a value greater than `~6e+36`, the function will underflow', function test( t ) {
	var r = rad2degf( 6.0e+36 );
```

File: lib/node_modules/@stdlib/math/base/special/rad2degf/test/test.native.js [hunk 10]
```diff
@@ -0,0 +1,174 @@
...
+			delta = abs( r - er );
+			tol = EPS * abs( er );
+			t.equal( delta <= tol, true, 'x: '+x[ i ]+'. r: '+r+'. expected: '+er+'. delta: '+delta+'. tol: '+ tol+'.' );
+		}
+	}
+	t.end();
+});
+
+tape( 'if provided a value greater than `~3.14e+38`, the function will underflow', opts, function test( t ) {
+	var r = rad2degf( 3.14e+38 );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/rad2degf/test/test.native.js]:
```suggestion
tape( 'if provided a value greater than `~6e+36`, the function will underflow', opts, function test( t ) {
	var r = rad2degf( 6.0e+36 );
```

File: lib/node_modules/@stdlib/math/base/special/rad2degf/README.md [hunk 11]
```diff
@@ -0,0 +1,200 @@
...
+
+<section class="notes">
+
+## Notes
+
+-   Due to finite precision, canonical values may not be returned. For example,
+
+    ```javascript
+    var d = rad2degf( 3.141592653589793 / 6.0 );
+    // returns 29.999999999999996
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/rad2degf/README.md]:
```suggestion
    // returns 30.000001907348633
```

File: lib/node_modules/@stdlib/math/base/special/rad2degf/docs/repl.txt [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+    > var d = {{alias}}( 3.141592653589793 / 2.0 )
+    90.0
+    > d = {{alias}}( -3.141592653589793 / 4.0 )
+    -45.0
+    > d = {{alias}}( NaN )
+    NaN
+
+    // Due to finite precision, canonical values may not be returned:
+    > d = {{alias}}( 3.141592653589793 / 6.0 )
+    29.999999999999996
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/rad2degf/docs/repl.txt]:
```suggestion
    30.000001907348633
```

**Review by kgryte — APPROVED** [review]

File: lib/node_modules/@stdlib/math/base/special/rad2degf/test/test.js (hunk 6 above)
Comment (gunjjoshi) [reply lib/node_modules/@stdlib/math/base/special/rad2degf/test/test.js]: Great, thanks!

File: lib/node_modules/@stdlib/math/base/special/rad2degf/README.md (hunk 11 above)
Comment (gunjjoshi) [reply lib/node_modules/@stdlib/math/base/special/rad2degf/README.md]: I would keep this in mind, to re-calculate values for `float32` versions, as they might be different than those for `float64`.

**Pull request merged**